app/controllers/predicates/AuthorisedAction.scala

```diff
@@ -44,21 +44,30 @@ class AuthorisedAction @Inject()(
   implicit val executionContext: ExecutionContext = cc.executionContext
 
 
-  def async(mtdItId: String)(block: User[AnyContent] => Future[Result]): Action[AnyContent] = defaultActionBuilder.async { implicit request =>
+  def async(block: User[AnyContent] => Future[Result]): Action[AnyContent] = defaultActionBuilder.async { implicit request =>
     implicit lazy val headerCarrier: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers)
-    authorised.retrieve(allEnrolments and affinityGroup) {
-      case enrolments ~ Some(AffinityGroup.Agent) =>
-        checkAuthorisation(block, enrolments, mtdItId, isAgent = true)(request, headerCarrier)
-      case enrolments ~ _ =>
-        checkAuthorisation(block, enrolments, mtdItId)(request, headerCarrier)
-    } recover {
-      case _: NoActiveSession =>
-        logger.info("AgentPredicate][authoriseAsAgent] - No active session. Redirecting to Unauthorised")
-        Unauthorized("")
-      case _: AuthorisationException =>
-        logger.info(s"[AgentPredicate][authoriseAsAgent] - Agent does not have delegated authority for Client.")
-        Unauthorized("")
-    }
+
+    request.headers.get("mtditid").fold {
+
+      logger.warn("[AuthorisedAction][async] - No MTDITID in the header. Returning unauthorised.")
+      Future.successful(Unauthorized(""))
```

Comment:
could we have a test for this case 

see Duanes PR he's got 
```
"the mtditid is not in the header" in {
          lazy val result = auth.async(block)(FakeRequest())
          status(result) mustBe UNAUTHORIZED
        }
```

Reply:
Added missing test